### PR TITLE
Add vercel_project_domain resource

### PR DESCRIPTION
**Pull request by lucasmarshall**
Required to add customer domains to our vercel metrics app project on customer provisioning.

Still need to figure out automating the build and including it in the terraform repo, but that is to come.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,14 +11,14 @@ test:
...
+	mkdir -p ~/.terraform.d/plugins/hashicorp.com/chronark/vercel/9000.1/darwin_amd64
+	mv terraform-provider-vercel ~/.terraform.d/plugins/hashicorp.com/chronark/vercel/9000.1/darwin_amd64
 
 
 fmt:
 
-	go generate -v ./...
-	golangci-lint run -v
+	# go generate -v ./...
+	# golangci-lint run -v
```
Comment (lucasmarshall) [inline Makefile:21]: These don't seem to work on the mac.

File: internal/provider/resource_project_domain.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+	if d.HasChange("redirect") {
+		update.Redirect = d.Get("redirect").(string)
+	}
+
+	projectDomain, err := client.ProjectDomain.Update(projectID, name, update, d.Get("team_id").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = d.Set("name", projectDomain.Name)
```
Comment (tomkit) [inline internal/provider/resource_project_domain.go:160]: is there a reason these are all set individually instead of at once?

**Review by tomkit — APPROVED** [review]

**Comment by tomkit** [comment]
synced in-person for this

File: internal/provider/resource_project_domain.go (hunk 2 above)
Comment (lucasmarshall) [reply internal/provider/resource_project_domain.go:160]: So we can check for errors for each one. Pretty standard go pattern.

**Pull request merged**